lib/grpnames.gd

```diff
@@ -119,10 +229,32 @@
 ##      This is done by merging the two sequences into
 ##   a single increasing sequence of pairs <c_i, which_i>
 ##   where which_i indicates where c_i is in the a-sequence
-##   and where it is in the b-sequence if any. The the
+##   and where it is in the b-sequence if any. Then the
 ##   linear algorithm above may be used.
 ##
-DeclareAttribute( "DirectFactorsOfGroup", IsGroup );
+DeclareGlobalFunction( "DirectFactorsOfGroupFromList",
+                        [ IsGroup, IsList, IsList ] );
+
+#############################################################################
+##
+#A  DirectFactorsOfGroupKN( <G> ) . . . . decomposition into a direct product
+##
+##  <ManSection>
+##  <Attr Name="DirectFactorsOfGroupKN" Arg="G"/>
+##
+##  <Description>
+##    A (sorted if possible) list of factors [<A>G1</A>, .., <A>Gr</A>] such
+##    that <A>G</A> = <A>G1</A> x .. x <A>Gr</A> and none of the <A>Gi</A>
+##    is a direct product.
+##    This is the implementation of the algorithm described in
+##    Neeraj Kayal and Timur Nezhmetdinov, Factoring Groups Efficiently,
+##    in International Colloquium on Automata, Languages and Programming
+##    (ICALP), Lecture Notes in Computer Science 5555, 585-596,
+##    Springer Verlag, Berlin Heidelberg 2009.
+##  </Description>
+##  </ManSection>
+##
+DeclareAttribute( "DirectFactorsOfGroupKN", IsGroup );
```

Comment:
Well, it at least makes more sense than adding a second attribute with more or less the same content (it takes up extra storage this way,  too).

So instead of an attribute, you would simpl define a function (via `BindGlobal`) which contains that code... And as I understand it, that function would not be called from anywhere right now -- unless the option is given, say "useKN", in which case your `DirectFactorsOfGroup` would call it..?!

If you do that, then reason for this ("it slower in my tests, but...") that should also be explained in comments, so that if somebody looks at that code in a year, they know the reason why certain things are as they are...